src/main/java/com/epam/aidial/cfg/service/ToolSetResourceService.java

```diff
@@ -151,8 +154,8 @@ public void deleteToolSetResource(String path,
 
     public McpSchema.ListToolsResult getDiscoveredTools(String path, String nextCursor) {
         var toolSet = getToolSetResource(path);
-        return toolDiscoveryService.discoverTools(toolSet.getEndpoint(),
+        return toolDiscoveryService.discoverTools(String.format(coreClientUrl + "/v1/toolset/%s/mcp", toolSet.getName()),
```

Comment:
1. handle case when coreClientUrl ends with trailing slash
2. `"/v1/toolset/%s/mcp"` for http transport and `"/v1/toolset/%s/sse"` for sse transport